### PR TITLE
added Error and Display impl for std::ffi::FromBytesWithNulError

### LINKED ISSUE #39925: `std::ffi::FromBytesWithNulError` is not an `std::error::Error`
It should be.

### PR DESCRIPTION
**Pull request by chordowl**
Fixes #39925.

This is my first PR, so I wasn't quite sure about the stability annotation.

### CODE REVIEW

File: src/libstd/ffi/c_str.rs [hunk 1]
```diff
@@ -455,6 +455,20 @@ impl From<NulError> for io::Error {
     }
 }
 
+#[unstable(feature = "frombyteswithnulerror_impls", reason = "recently added", issue = "39925")]
```
Comment (alexcrichton) [inline src/libstd/ffi/c_str.rs]: Oh unfortunately impls aren't subject to stability the same way as other parts of the standard library, so these'll be "insta-stable" so it's ok to just go ahead and tag this with `#[stable]`

**Comment by alexcrichton** [comment]
Looks good to me, thanks @lukaramu!

@rfcbot fcp merge

File: src/libstd/ffi/c_str.rs [hunk 2]
```diff
@@ -455,6 +455,20 @@ impl From<NulError> for io::Error {
...
+impl Error for FromBytesWithNulError {
+    fn description(&self) -> &str {
+        "data provided is not null terminated or contains an interior nul byte"
```
Comment (ArtBears) [inline src/libstd/ffi/c_str.rs]: Shouldn't it be interior *null* byte?

Comment (jmesmon) [inline src/libstd/ffi/c_str.rs]: Not necessarily. https://en.wikipedia.org/wiki/Null_character

Comment (ArtBears) [inline src/libstd/ffi/c_str.rs]: Ah I see, thanks for the correction @jmesmon

**Pull request merged**